buffer.c

```diff
@@ -351,3 +351,29 @@ buffer_print(buffer_t *self) {
 
   printf("\n");
 }
+
+/*
+ * Checks if the buffer's end is same than `ends`.
```

Comment:
Documentation should use the imperative mood, plus a few grammatical issues:

```suggestion
 * Check if the provided buffer ends with the specified `ends` character array.
```